### PR TITLE
fix: Make ‘resolveStyles’ support deep objects, add ‘resolveClassName’

### PR DESCRIPTION
**Pull request by markdalgleish**
Also updated the readme to showcase these new APIs, and to better highlight basic non-themed usage.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -23,7 +23,9 @@ Because theming is achieved by generating multiple classes, **legacy browsers ar
 ---
 
 - [Requirements](#requirements)
-- [tl;dr (React version)](#tldr-react-version)
+- [Example Usage](#example-usage)
+  - [Basic Usage](#basic-usage)
+  - [Themed Usage (React version)](#themed-usage-react-version)
```
Comment (mattsjones) [inline README.md]: Bit weird that we only mention React in the theme stuff but the basic usage is React as well. Not sure if that's confusing.

File: packages/treat/tests/fixtures/react/Demo.js [hunk 2]
```diff
@@ -1,16 +1,21 @@
 import React from 'react';
```
Comment (mattsjones) [inline packages/treat/tests/fixtures/react/Demo.js]: Not your fault but this file is no longer used. Just remove it.

File: README.md (hunk 1 above)
Comment (markdalgleish) [reply README.md]: I'd be keen to use a non-React demo for the basic usage example. Hard to make it readable, though. Any ideas?

Comment (mattsjones) [reply README.md]: Maybe ejs or something? Feels kinda dated though.

Comment (markdalgleish) [reply README.md]: Could just make the example create an HTML template literal, not an entire component.

File: packages/treat/tests/fixtures/react/Demo.js (hunk 2 above)
Comment (markdalgleish) [reply packages/treat/tests/fixtures/react/Demo.js]: Was sure I did this... but I guess not 🤷‍♂

**Pull request merged**